lib/internal/url.js

```diff
@@ -67,60 +67,160 @@ class TupleOrigin {
   }
 
   toString(unicode = false) {
-    var result = this.scheme;
+    var result = this[kScheme];
     result += '://';
-    result += unicode ? URL.domainToUnicode(this.host) : this.host;
-    if (this.port !== undefined && this.port !== null)
-      result += `:${this.port}`;
+    result += unicode ? URL.domainToUnicode(this[kHost]) : this[kHost];
+    if (this[kPort] !== undefined && this[kPort] !== null)
+      result += `:${this[kPort]}`;
     return result;
   }
 
   inspect() {
     return `TupleOrigin {
-      scheme: ${this.scheme},
-      host: ${this.host},
-      port: ${this.port},
-      domain: ${this.domain}
+      scheme: ${this[kScheme]},
+      host: ${this[kHost]},
+      port: ${this[kPort]},
+      domain: ${this[kDomain]}
     }`;
   }
 }
 
+function onParseComplete(flags, protocol, username, password,
+                         host, port, path, query, fragment) {
+  if (flags & binding.URL_FLAGS_FAILED)
+    throw new TypeError('Invalid URL');
+  const ctx = this[context];
```

Comment:
I wouldn't say that we *have* to do this. This is more a code hygiene thing for me to avoid duplicated accessor calls but beyond that the impact is negligible.